conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
-        self.assertEqual(cppstd_default("clang", "5"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "5.1"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "6"), "gnu14")
-        self.assertEqual(cppstd_default("clang", "7"), "gnu14")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "2.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "2.1")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "3.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "3.1")
+        self.assertEqual(make_cppstd_default("clang", "3.4"), "gnu98")
+        self.assertEqual(make_cppstd_default("clang", "3.5"), "gnu98")
+        self.assertEqual(make_cppstd_default("clang", "5.0"), "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "5.1")
+        self.assertEqual(make_cppstd_default("clang", "6.0"), "gnu14")
+        self.assertEqual(make_cppstd_default("clang", "7.0"), "gnu14")
 
     def test_apple_clang_cppstd_flags(self):
-        self.assertEqual(cppstd_flag("apple-clang", "3.9", "98"), None)
-        self.assertEqual(cppstd_flag("apple-clang", "3.9", "gnu98"), None)
-        self.assertEqual(cppstd_flag("apple-clang", "3.9", "11"), None)
-        self.assertEqual(cppstd_flag("apple-clang", "3.9", "14"), None)
-
-        self.assertEqual(cppstd_flag("apple-clang", "4.0", "98"), "-std=c++98")
-        self.assertEqual(cppstd_flag("apple-clang", "4.0", "gnu98"), "-std=gnu++98")
-        self.assertEqual(cppstd_flag("apple-clang", "4.0", "11"), "-std=c++11")
-        self.assertEqual(cppstd_flag("apple-clang", "4.0", "14"), None)
-
-        self.assertEqual(cppstd_flag("apple-clang", "5.0", "98"), "-std=c++98")
-        self.assertEqual(cppstd_flag("apple-clang", "5.0", "gnu98"), "-std=gnu++98")
-        self.assertEqual(cppstd_flag("apple-clang", "5.0", "11"), "-std=c++11")
-        self.assertEqual(cppstd_flag("apple-clang", "5.0", "14"), None)
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "3.9", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "3.9", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "3.9", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "3.9", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "4.0", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "4.0", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "4.0", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("apple-clang", "4.0", "14")
```

Comment:
```suggestion
```

These are failing because of the `settings.yml` and the helper function, not because of the `cppstd_flag`.